lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -265,67 +265,207 @@ function _isAccountId(principal) {
     return (principal.length === 12 && /^\d+$/.test(principal));
 }
 
-function _checkPrincipal(requester, principal) {
-    if (principal === '*') {
-        return true;
-    }
-    // User in unauthenticated (anonymous request)
-    if (requester === undefined) {
+function _isRootUser(arn) {
+    if (!arn) {
         return false;
     }
-    if (principal === requester) {
+
+    // Vault returns the following arn when the account makes requests 'arn:aws:iam::295412255825:/master/',
+    // with an empty resource type ('user/' prefix missing).
+    const arns = arn.split(':');
+    const resource = arns.at(-1);
```

Comment:
Just a not on the `.at`, it seems this is quite recent (es2022), as you target the 7.70 branch, we will want to be careful if the change is later backported to a hotfix branch, that might still use an older nodejs version, as if I'm not mistaken, this one was introduced in later v16 nodejs version, it's definitely not in the first v16 versions.